components/Link.js

```diff
@@ -1,14 +1,31 @@
 import { html } from 'https://unpkg.com/rplus-production@1.0.0';
 import shouldPreventDefault from '../utils/shouldPreventDefault.js';
 
-export default ({ href, onClick, innerText }) => {
+const pushState = url => history.pushState(null, null, url);
+
+/** Link component props:
+ * href = anchor href
+ * innerText = text to display between anchor tags
+ * customOnClick = a function (or boolean) that:
```

Comment:
How come you would use a `Link` when you want custom click behaviour and not just a `button`?

Reply:
See my comment below (quoted):

> `noPreventDefault` in combination with `customOnClick` (i.e. both of these being `true`) would mean nothing runs `onClick` and make it act like a typical `a`, yes!

`customOnClick` could:
a) make it simply act like an anchor tag (if we pass in `true`)
b) provide a function that we still *might* want to include on what otherwise acts as an anchor (e.g. like what we're doing with `pushState`)

If you think `b)` is unnecessary/unlikely to be used, I could change this to `disableOnClick` and just use it for option `a)`?

Reply:
My point is that if you want just a normal link with page reloading then I'd prefer people just to use an `a` tag and if they want just an onClick with no navigation then they should use a `button`. Single responsibility components like `Link` usually encapsulate one specific behaviour (changing the url without reloading the page in this case). Keeping it simple like that generally prevents misuse.